### PR TITLE
Update pre-commit hook versions

### PR DESCRIPTION
**Pull request by Borda**
Running regular update `pre-commit autoupdate`...

Moreover curious if it would be possible to install this [bot](https://github.com/marketplace/pre-commit-ci), which would help with fixing open PRs and do these regular updates automatically (via PR so all is validated before merge)

```yaml
ci:
  autofix_prs: true
  autoupdate_commit_msg: "[pre-commit.ci] pre-commit suggestions"
  autoupdate_schedule: quarterly
  # submodules: true
```

cc: @AA-Turner @hugovk

### CODE REVIEW

**Review by hugovk — APPROVED** [review]
+1 to https://pre-commit.ci

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 repos:
```
Comment (Borda) [inline .pre-commit-config.yaml:1]:
```suggestion
ci:
  autofix_prs: true
  autoupdate_commit_msg: "[pre-commit.ci] pre-commit suggestions"
  autoupdate_schedule: quarterly
  # submodules: true

repos:
```

Comment (hugovk) [reply .pre-commit-config.yaml:1]: `autofix_prs: true` is the default and can be omitted. 

I usually don't change the commit message either, but no strong opinion on that.

`autoupdate_schedule: quarterly` is good, much less noisy than weekly!

Remove `# submodules: true` if not used.

Comment (Borda) [reply .pre-commit-config.yaml:1]: > `autofix_prs: true` is the default and can be omitted.

True, but I think it is better to repeat it so everyone is aware of it :)

**Comment by Borda** [comment]
@pradyunsg @GaretJax could you pls have look :)

**Comment by Borda** [comment]
friendly ping @pradyunsg @GaretJax if you could have a look :chipmunk:

**Comment by Borda** [comment]
> friendly ping @pradyunsg @GaretJax if you could have a look 🐿️

mind have a look, when you have time 🦩

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,6 +1,11 @@
+ci:
+  autofix_prs: true
+  autoupdate_commit_msg: "[pre-commit.ci] pre-commit suggestions"
+  autoupdate_schedule: quarterly
+
```
Comment (AA-Turner) [inline .pre-commit-config.yaml:5]:
```suggestion
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
 repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.3.7
+    rev: v0.4.4
```
Comment (AA-Turner) [inline .pre-commit-config.yaml:8]:
```suggestion
    rev: v0.5.7
```

**Comment by Borda** [comment]
@AA-Turner thank you, I just landed your suggestion :)

**Pull request merged**